p2p/discovery/database.go

```diff
@@ -34,10 +35,19 @@ func (db *database) find(id common.Hash) *Node {
 	return db.m[id]
 }
 
-func (db *database) delete(id common.Hash) {
+func (db *database) delete(id *common.Hash) {
 	db.mutex.Lock()
 	defer db.mutex.Unlock()
-	delete(db.m, id)
+	delete(db.m, *id)
+}
+
+func (db *database) size() int {
+	db.mutex.Lock()
+	defer db.mutex.Unlock()
+
+	size := len(db.m)
+
+	return size
```

Comment:
merge into return len(db.m) seems would be clean.

Reply:
sure